### ISSUE TITLE
XML compare

### ISSUE THREAD

**Issue by vanjikumaran** [ballerina-platform/ballerina-lang#9736]
**New Feature Description**
XML compare, Since the ballerina, contains the XML as a part of it native datatype. Comparing 2 XML variable should be a native function in a ballerina. If the same elements and values are part of variables. It should return true regardless of the order of elements

### Test Case 1
```
Target = <RootElement><ChildElement1>Value1</ChildElement1><ChildElement2>value2</ChildElement2></RootElement>
Source = <RootElement><ChildElement1>Value1</ChildElement1><ChildElement2>value2</ChildElement2></RootElement>
```

### Test Case 2
```
Target = <RootElement><ChildElement1>Value1</ChildElement1><ChildElement2>value2</ChildElement2></RootElement>
Source = <RootElement><ChildElement2>Value1</ChildElement2><ChildElement1>value2</ChildElement1></RootElement>
```
### Test Case 3
```
Target = <RootElement><ChildElement1>ValueWrong</ChildElement1><ChildElement2>value2</ChildElement2></RootElement>
Source = <RootElement><ChildElement2>Value1</ChildElement2><ChildElement1>value2</ChildElement1></RootElement>
```

### Test Case 4
```
Target = <RootElement><ChildElement1>Value1</ChildElement1><ChildElement2>value2</ChildElement2></RootElement>
Source = <RootElement><ChildElement2>Value1</ChildElement2><ChildElement1>value2</ChildElement1></RootElement>
```

**Comment by SupunS** [comment]
Thanks @vanjikumaran for the suggestion. I think this is something that needs to implemented as the behavior of the `==` for XML.

The challenge we have is to compare the XML in its canonical form (https://www.w3.org/TR/xml-c14n11/). We are yet to explore how we can handle this in ballerina. Once we have a way to canonicalize XML, then this would be very straight forward to implement.

**Comment by MaryamZi** [comment]
Introduced with https://github.com/ballerina-platform/ballerina-lang/pull/10904, and will be available from 0.990.0 onwards.

For two XML values to be equal, even though attribute order does not matter, the element order is taken into consideration.
